### PR TITLE
[coqdep] Make some options and API internal

### PR DESCRIPTION
**Pull request by ejgallego**
This is preparation to move coqdep to use a common loadpath / coqlib
library.

### CODE REVIEW

File: tools/coqdep.ml [hunk 1]
```diff
@@ -8,84 +8,10 @@
...
-  | "-dyndep" :: "opt" :: ll -> option_dynlink := Opt; parse ll
-  | "-dyndep" :: "byte" :: ll -> option_dynlink := Byte; parse ll
-  | "-dyndep" :: "both" :: ll -> option_dynlink := Both; parse ll
-  | "-dyndep" :: "var" :: ll -> option_dynlink := Variable; parse ll
-  | ("-h"|"--help"|"-help") :: _ -> usage ()
-  | opt :: ll when String.length opt > 0 && opt.[0] = '-' ->
-     coqdep_warning "unknown option %s" opt; parse ll
-  | f :: ll -> treat_file None f; parse ll
-  | [] -> ()
+let (//) = Filename.concat
```
Comment (SkySkimmer) [inline tools/coqdep.ml:14]: Why not `(/)`? I don't think we use integer division in coqdep.

File: tools/coqdep_common.ml [hunk 2]
```diff
@@ -11,6 +11,8 @@
 open Format
 open Unix
 open Coqdep_lexer
+(* The whole Minisys module used to be required due to bootstrapping,
+   thanks to dune it is not needed anymore and we could just remove it. *)
```
Comment (SkySkimmer) [inline tools/coqdep_common.ml:15]: Why don't we remove it then? Or do you prefer to do that in the next PR?

**Review by SkySkimmer — APPROVED** [review]

File: tools/coqdep.ml (hunk 1 above)
Comment (ejgallego) [reply tools/coqdep.ml:14]: I kept `//` as it was used already in the code with this particular format.

File: tools/coqdep_common.ml (hunk 2 above)
Comment (ejgallego) [reply tools/coqdep_common.ml:15]: Yup, I guess I will take care of this one in #14059

**Comment by ejgallego** [comment]
Ping @SkySkimmer

**Comment by SkySkimmer** [comment]
@coqbot: merge now

**Pull request merged**